src/HospitalModel.H

```diff
@@ -211,35 +171,49 @@ void HospitalModel<PCType, PTDType, PType>::treatAgents(PCType& a_agents, /*!< A
                 Gpu::synchronize();
             }
 
+            bool is_census = (a_agents.ic_type == ExaEpi::ICType::Census);
+            auto grid_to_lnglat_ptr = &a_agents.grid_to_lnglat;
+
             ParallelFor( np, [=] AMREX_GPU_DEVICE (int i) noexcept
             {
                 if ( !isHospitalized(i, ptd) )  { return; }
 
                 if (is_alive_ptr[i] == 0) {
-
                     // agent has died
                     for (int d = 0; d < n_disease; d++) {
                         status_ptrs[d][i] = Status::dead;
                     }
                     hosp_i_ptr[i] = -1;
                     hosp_j_ptr[i] = -1;
                     withdrawn_ptr[i] = 0;
-
                 } else {
-
                     // check if agent can be discharged from hospital
                     ParticleReal sum_timers = 0;
                     for (int d = 0; d < n_disease; d++) {
                         sum_timers += timer_ptrs[d][i];
                     }
-                    if (sum_timers == 0) {
+                    bool is_infected = false;
+                    for (int d = 0; d < n_disease; d++) {
+                        if (status_ptrs[d][i] == Status::infected) {
+                            is_infected = true;
+                            break;
+                        }
+                    }
+                    if (sum_timers == 0 || !is_infected) {
```

Comment:
Why do we need to check whether the agent is still infected here? If we are at this point in the code, then we considering agents in a hospital whose timers have counted down to the end of hospital duration and are not dead. So, their status must have already been set to `Status::immune` earlier, right?

Reply:
Agreed. Fixed.